akka-docs/rst/scala/http/introduction.rst

```diff
@@ -10,23 +10,105 @@ You get to pick the API level of abstraction that is most suitable for your appl
 This means that, if you have trouble achieving something using a high-level API, there's a good chance that you can get
 it done with a low-level API, which offers more flexibility but might require you to write more application code.
 
+DSL for HTTP servers
+--------------------
+The high-level, routing API of Akka HTTP provides a DSL to describe HTTP "routes" and how they should be handled.
+Each route is composed of one or more level of ``Directive`` s that narrows down to handling one specific type of
+request.
+
+For example one route might start with matching the ``path`` of the request, only matching if it is "/hello", then
+narrowing it down to only handle HTTP ``get`` requests and then ``complete`` those with a Scala XML literal, which
```

Comment:
"object for which an implicit marshaller is provided" "default marshallers are provided for simple objects like String or ByteString, and you can define your own for example for json etc (links here)